### DISCUSSION TITLE
Enhance `mkdwarfs` to support something like "same folder similarity"

### DISCUSSION THREAD

**Discussion by cipriancraciun** [mhx/dwarfs#8] [Ideas]
(This ticket is more to spark a discussion, than anything actually actionable without further experimentation.)

I understand that the similarity clustering feature is the best "global" option.  However, there are perhaps use-cases, when something in-between "path" and "similarity" works best.

For example if I would to deploy a large Python application, or a large dataset to be exported via HTTP, most of the time if one accesses a file in a given folder, than most likely other files in the same folder would be also accessed.

Thus a nice option would be to have something like `path-similarity` that first sorts all files in the path order (I think breath first would work best) and then tries to apply similarity within a certain window.

**Comment by mhx** [comment]
Thinking about this, what I'll probably do is to expose the similarity hash for each file through the Lua interface. Then you can basically implement whatever ordering scheme you like based on path/file name, attributes and similarity.
